src/language.rs

```diff
@@ -30,161 +30,148 @@ pub enum SupportedLanguageName {
-        SupportedLanguage {
-            language: tree_sitter_elisp::language(),
-            name: SupportedLanguageName::Elisp,
-            name_for_ignore_select: "elisp",
-        },
-        SupportedLanguage {
-            language: tree_sitter_elm::language(),
-            name: SupportedLanguageName::Elm,
-            name_for_ignore_select: "elm",
-        },
-        SupportedLanguage {
-            language: tree_sitter_dockerfile::language(),
-            name: SupportedLanguageName::Dockerfile,
-            name_for_ignore_select: "docker",
-        },
-        SupportedLanguage {
-            language: tree_sitter_html::language(),
-            name: SupportedLanguageName::Html,
-            name_for_ignore_select: "html",
-        },
-        SupportedLanguage {
-            language: tree_sitter_query::language(),
-            name: SupportedLanguageName::TreeSitterQuery,
-            name_for_ignore_select: "treesitterquery",
-        },
-        SupportedLanguage {
-            language: tree_sitter_json::language(),
-            name: SupportedLanguageName::Json,
-            name_for_ignore_select: "json",
-        },
-        SupportedLanguage {
-            language: tree_sitter_css::language(),
-            name: SupportedLanguageName::Css,
-            name_for_ignore_select: "css",
-        },
-        SupportedLanguage {
-            language: tree_sitter_lua::language(),
-            name: SupportedLanguageName::Lua,
-            name_for_ignore_select: "lua",
-        },
+pub static ALL_SUPPORTED_LANGUAGES: BySupportedLanguage<SupportedLanguage> = {
+    use SupportedLanguage::*;
+    [
+        Rust,
+        Typescript,
+        Javascript,
+        Swift,
+        ObjectiveC,
+        Toml,
+        Python,
+        Ruby,
+        C,
+        Cpp,
+        Go,
+        Java,
+        CSharp,
+        Kotlin,
+        Elisp,
+        Elm,
+        Dockerfile,
+        Html,
+        TreeSitterQuery,
+        Json,
+        Css,
+        Lua,
+    ]
+};
+
+static SUPPORTED_LANGUAGE_LANGUAGES: Lazy<BySupportedLanguage<Language>> = Lazy::new(|| {
+    [
+        tree_sitter_rust::language(),
+        tree_sitter_typescript::language_tsx(),
+        tree_sitter_javascript::language(),
+        tree_sitter_swift::language(),
+        tree_sitter_objc::language(),
+        tree_sitter_toml::language(),
+        tree_sitter_python::language(),
+        tree_sitter_ruby::language(),
+        tree_sitter_c::language(),
+        tree_sitter_cpp::language(),
+        tree_sitter_go::language(),
+        tree_sitter_java::language(),
+        tree_sitter_c_sharp::language(),
+        tree_sitter_kotlin::language(),
+        tree_sitter_elisp::language(),
+        tree_sitter_elm::language(),
+        tree_sitter_dockerfile::language(),
+        tree_sitter_html::language(),
+        tree_sitter_query::language(),
+        tree_sitter_json::language(),
+        tree_sitter_css::language(),
+        tree_sitter_lua::language(),
```

Comment:
This would obviously be sketchy as far as not messing up the order if the idea wasn't that it would be macro-generated by something where "the relevant things are grouped together"

I guess I decided to make these + the `name_for_ignore_select`'s their own separate data structures (vs in the existing code they are fields on the `SupportedLanguage` struct) since that seemed like the simplest way to "free up" `SupportedLanguage` to just be a simple "token" enum

But might be interesting to explore how you'd represent the "main type that's getting passed around" (eg `SupportedLanguage`) as something with a uniform set of fields too